src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -2,77 +2,60 @@
     /// </summary>
     internal sealed class ParsedStackFrame : ParsedFrame
     {
+        public readonly StackFrameTree Tree;
+
         public ParsedStackFrame(
-            string originalText,
-            TextSpan typeSpan,
-            TextSpan methodSpan,
-            TextSpan argsSpan,
-            TextSpan fileSpan = default)
-            : base(originalText)
+            StackFrameTree tree)
         {
-            Contract.ThrowIfTrue(typeSpan.IsEmpty);
-            Contract.ThrowIfTrue(methodSpan.IsEmpty);
-
-            TypeSpan = typeSpan;
-            MethodSpan = methodSpan;
-            ArgsSpan = argsSpan;
-            FileSpan = fileSpan;
+            Tree = tree;
         }
 
-        /// <summary>
-        /// The full type name parsed from the line. 
-        /// ex: [|Microsoft.CodeAnalysis.Editor.CallstackExplorer.|]Example(arg1, arg2)
-        /// </summary>
-        public TextSpan TypeSpan { get; }
-
-        /// <summary>
-        /// The method name span
-        /// ex: Microsoft.CodeAnalysis.Editor.CallstackExplorer.[|Example|](arg1, arg2)
-        /// </summary>
-        public TextSpan MethodSpan { get; }
+        public StackFrameCompilationUnit Root => Tree.Root;
 
-        /// <summary>
-        /// The span of comma seperated arguments.
-        /// ex: Microsoft.CodeAnalysis.Editor.CallstackExplorer.Example[|(arg1, arg2)|]
-        /// </summary>
-        public TextSpan ArgsSpan { get; }
-
-        /// <summary>
-        /// The span representing file information on the stack trace line. Is not always available, so it's 
-        /// possible this span is <see langword="default"/>
-        /// </summary>
-        public TextSpan FileSpan { get; }
+        public override string ToString()
+        {
+            return Tree.Text.CreateString();
+        }
 
         public async Task<ISymbol?> ResolveSymbolAsync(Solution solution, CancellationToken cancellationToken)
         {
-            // The original span for type includes the trailing '.', which we don't want when
-            // looking for the class by metadata name
-            var fullyQualifiedTypeName = OriginalText[TypeSpan.Start..(TypeSpan.End - 1)];
-            var methodName = GetMethodText();
+            // MemberAccessExpression is [Expression].[Identifier], and Identifier is the 
+            // method name.
+            var typeExpression = Root.MethodDeclaration.MemberAccessExpression.Left;
+            var fullyQualifiedTypeName = typeExpression.ToString(skipTrivia: true);
+
+            RoslynDebug.AssertNotNull(fullyQualifiedTypeName);
+
+            var methodIdentifier = Root.MethodDeclaration.MemberAccessExpression.Right;
+            var methodTypeArguments = Root.MethodDeclaration.TypeArguments;
+            var methodArguments = Root.MethodDeclaration.ArgumentList;
+
+            var methodName = methodIdentifier.ToString(skipTrivia: true);
 
             foreach (var project in solution.Projects)
```

Comment:
for the future: consider having hte resolution sit at a higher layer, being passed a lsit of stack frames.  teh reason for this is that if we are able to resolve a frame in one project, then successive frames will likely resolve in that project as well.  we can then be faster/smarter about whihc projects to look at to get the overall result.

Reply:
You mean like putting it on `StackTraceAnalyzer` static class? I wouldn't mind moving it